### ISSUE TITLE
Custom sentence boundaries (token.is_sent_start = True) broken

### ISSUE THREAD

**Issue by bittlingmayer** [explosion/spaCy#7214]
As part of upgrading to 3.0.3, we have to change how we add custom pipeline components.

In our case, we want to split sentences on certain character sequences like `'...'` and `'  '`, which is covered by an example in the docs.

(It was working for us under the old regime.)

## How to reproduce the behaviour

Use the code in https://spacy.io/usage/linguistic-features#sbd-custom and add print statements:
```
from spacy.lang.en import English
from spacy.language import Language

@Language.component('set_custom_boundaries')
def set_custom_boundaries(doc):
    for token in doc[:-1]:
        if token.text == '...':
            print('Splitting before:', doc[token.i + 1])
            doc[token.i + 1].is_sent_start = True
    return doc

nlp = English()
nlp.add_pipe('sentencizer')
nlp.add_pipe('set_custom_boundaries', before='sentencizer')

doc = nlp('this is a sentence. hello. and another sentence.')
for sent in doc.sents:
    print(sent)

doc = nlp('this is a sentence...hello...and another sentence.')
for sent in doc.sents:
    print(sent)
```
The actual output will be:
```
this is a sentence.
hello.
and another sentence.
Splitting before: hello
Splitting before: and
this is a sentence...hello...and another sentence.
```
The expected output would be:
```
...
this is a sentence...
hello...
and another sentence.
```

## Your Environment

* Operating System: MacOS
* Python Version Used: 3.7
* spaCy Version Used: 3.0.3
* Environment Information:

**Comment by bittlingmayer** [comment]
This is time-sensitive, so we'd be happy for any workarounds we can use now until a fix is out.

**Comment by bittlingmayer** [comment]
Seems like this is the issue:

> Important note

> To prevent inconsistent state, you can only set boundaries before a document is parsed (and doc.has_annotation("DEP") is False). To ensure that your component is added in the right place, you can set before='parser' or first=True when adding it to the pipeline using nlp.add_pipe

After removing `before='sentencizer'`, the output is as expected.

**Comment by adrianeboyd** [comment]
This will probably fixed by #7176. The `Sentencizer.__call__` method ended up in an incorrect state after some v3 refactoring, and it clobbers existing boundaries when `Sentencizer.pipe` doesn't.

A workaround is to use `nlp.pipe([text])` instead, or to add your component after `sentencizer` instead of before it.
